src/WaveTrack.cpp

```diff
@@ -201,13 +188,8 @@ void WaveTrack::Reinit(const WaveTrack &orig)
          mpSpectrumSettings.reset();
    }
 
-   {
-      auto &settings = orig.mpWaveformSettings;
-      if (settings)
-         mpWaveformSettings = std::make_unique<WaveformSettings>(*settings);
-      else
-         mpWaveformSettings.reset();
-   }
+   // Copy attached data from orig.  Nullify data in this where orig had null.
+   ((Caches&)*this) = orig;
```

Comment:
Why C-cast?

Reply:
Invoke assignment only if the ClientData::Site base class to clone the attachments.  That conserves the effect of the deleted lines.

Reply:
I do not understand what exactly do you mean and how it justifies C-cast.

Reply:
More explicitly `this->Caches::operator=(orig)` with no cast but I don’t find that prettier.

This is an intentional slicing assignment.

Reply:
I don't find this explanation convincing. This should be solved differently. Definitely not with a C-cast. 

I don't quite understand, why `static_cast` or some form of implicit conversion is not used here. 

I am not going to approve the code in the way it is now done.

Reply:
Okay then, will you be satisfied with:

`static_cast<Caches&>(*this) = orig;` ?

The reason C-cast didn't bother me is that this is an up-cast of the object to its base class, not the checked down-cast which is what `static_cast` is usually for with pointers and references.  In other words this also works with no casts:

```
Caches &caches = *this;
caches = orig;
```

In fact I even like that better.  No casts, just a reference binding.

Reply:
As to what the line actually DOES, never mind the style matters -- there is no complaint?  It does no more nor less than the deleted lines it replaces.

Reply:
Implicit conversion, as with two lines you proposed, works just fine for me.